stan/math/gpu/matrix_gpu.hpp

```diff
@@ -16,7 +17,12 @@
  */
 namespace stan {
 namespace math {
+const int Lower = 0;
```

Comment:
Can you also put in a comment here and in the kernel that these must match? What I'm afraid of is that someone see this and thinks "oh, I can change this" without realizing that it's actually duplicated in the kernel.

Alternatively, is there a different way to do it so the constants are shared in some way?

Reply:
I changed this so that the the constants for the kernels are set as flags at compile time (see: https://github.com/stan-dev/math/pull/916/commits/0a2df8501088f39d54352d05ff294560475fe5d0#diff-23051186f20ae937b29d69ce3b413239R238). This way if anyone changes the const this is changed in both places. 

Due to this I had to move the constants to the opencl_context.hpp file. Is that ok? If not, I can go switch back and just add the docs as you suggested.

I also left the #ifndefs in the kernels (https://github.com/stan-dev/math/pull/916/commits/0a2df8501088f39d54352d05ff294560475fe5d0#diff-ce4c3c3e53a52143616423ff7eb6c82aR10) if anyone wants to just use the kernel.

Reply:
Awesome! I definitely wouldn't have even envisioned a fix that like! Thanks... I really think that helps from someone accidentally breaking everything.